### ISSUE TITLE
Tips don't disappear when elements do

### ISSUE THREAD

**Issue by jlecour** [jaz303/tipsy#19]
Hi,

I have a table of data. On some items, there is an associated tipsy. The content of the table is periodicaly updated (the content of a specific tag is replaced).

Sometimes, an activated tipsy stays visible after the content replacement. As a result I have several zombie tipsies on my page.

Looking at the source code, I can see that I have many divs created by Tipsy.

I've tried to use the "live" option. It allows me not to rebind tipsy to my items after each update, but it doesn't solve this issue.

Is it something normal (as in "an edge case that I'm the only one to have") or I'm using it the wrong way ?
Anyway, I'd appreciate some directions.

Thank you for this great and very useful jquery plugin

**Comment by jlecour** [comment]
I've just found something.

Before updating my table content, I remove all the tipsies : `$(".tipsy").remove();`
It seems to work.

It's not ideal, because If I hover an element with a tipsy, and don't move my pointer, each time my content is updated the tipsy disappear. I have to move my pointer a bit to make it appear again.

**Comment by ghost** [comment]
I can confirm this issue. There should be a way to hook into jquery to remove tipsy() elements when their trigerring elements are removed. But other tooltip libraries don't do this, so its possible.

**Comment by jaz303** [comment]
Yep I'd like to fix this. Do you have any examples of said tooltip libs? Tipsy appends tooltips to `document.body` - I suspect other libs might append the tooltip to the pointee.

Cheers,

-J

**Comment by steve-obrien** [comment]
Good example to create zombie tooltips:  

Create a button that ajax's a replacement of itself. With a live tipsy on it. When it ajax's in, the old tooltip stays there and looses its events.  Quite tricky to explain.

**Comment by iwasrobbed** [comment]
Using `$(".tipsy").remove();` seems to do the trick since a div with a "tipsy" class is appended to the body of the doc when the tooltip is shown.  

Just be sure you don't use a "tipsy" class elsewhere (i.e. call your tooltip's something like "toolTip")

**Comment by deanc** [comment]
+1 vote for this. Came across this issue today.

**Comment by chandlermania** [comment]
I was able to work around this by triggering a "mouseout" event before my AJAX call.  Looking in tipsy,js, it closes the hover after 100 milliseconds, so I added a timeout of 150 to give it time to close before replacing the element and running AJAX call. Zombie tooltips went away ;)

<code>
function updateBoolean(where, what) {
    var field_name = $(what).attr("name");
    if($(what).is(":checked")) {
        var value=1;
    } else {
        var value=0;
    }
    var loading = $("<img src='images/ajax_loader_16.gif' id='load_"+field_name+"' hspace='2' alt='saving' />");
    $(what).trigger("mouseout");  
    setTimeout(function(){
        $(what).replaceWith(loading);
        $.ajax({
            type: "POST",
            url: "#####################",
            data: "##############################",  
            cache: false,
            success: function(){
                loading.replaceWith($(what));
            }
         });},
    150);

}
</code>

**Comment by costa** [comment]
https://github.com/jaz303/tipsy/pull/77

**Comment by ghost** [comment]
Was able to bypass this issue by calling 

`````` $('a[rel=tipsy]').trigger("mouseout");```
just before sending the AJAX request ($.ajax->beforeSend)
and then adding back the tipsy "engine" on the page on response
  ```$('a[rel=tipsy]').tipsy({fade: true, gravity: 'w'});```
``````
